### PR TITLE
Yonah arm64 backend updates

### PR DESCRIPTION
**Pull request by YonahGoldberg**
Arm64 backend refactoring to pave way for assembler support.
- added arm64-linux target
- removed arm64-darwin target
- added class Arm64Linux that inherits from class Arm64Backend
- compiled programs still give InternalError "not implemented"

### CODE REVIEW

File: aeneas/src/arm64/Arm64Backend.v3 [hunk 1]
```diff
@@ -1,38 +1,31 @@
...
+component Arm64Common {
+	def LOWERING = makeMachLoweringConfig();
+	def makeMachLoweringConfig() -> MachLoweringConfig {
+		var config = MachLoweringConfig.new();
+		config.Int32Arith = true;
+		config.Int64Arith = true;
+		config.ExplicitDivChecks = true;
+		config.ExplicitModChecks = true;
+		config.IntConvertFUnsigned = false; // no unsigned conversions until AVX?
+		config.IntConvertFMapsNanToZero = false; // cvts{s,d}2si maps NaN to int.min
```
Comment (titzer) [inline aeneas/src/arm64/Arm64Backend.v3:16]: I think these comments aren't accurate for arm.

File: aeneas/src/arm64/Arm64Backend.v3 [hunk 2]
```diff
@@ -49,43 +42,58 @@ class Arm64Backend extends MachBackend {
...
+		if (compiler.useGlobalRegAllocMatcher != VstMatcher.None) allocateRegsGlobal = GlobalRegAlloc.new(MRegs.SET, codegen).allocate;
+		if (compiler.LocalRegAlloc) allocateRegs = LocalRegAlloc.new(MRegs.SET, codegen).allocate;
+		else allocateRegs = SimpleRegAlloc.new(MRegs.SET, codegen).allocate;
 	}
 
 	// Override MachBackend	
+
 	def genEntryStub() {
-		prog.ERROR.unimplemented();
+		mach.fail("not implemented");
```
Comment (titzer) [inline aeneas/src/arm64/Arm64Backend.v3:55]: I think it'd be helpful to introduce a method that is `unimplemented()` to save a little redundancy.

**Review by titzer — APPROVED** [review]
Mostly looks good.

**Comment by titzer** [comment]
This will need to be rebased on the newest changes.

File: aeneas/src/arm64/Arm64Linux.v3 [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	new(compiler: Compiler, prog: Program, mach: MachProgram, w: MachDataWriter, dwarf: Dwarf,
+		test: bool) super(compiler, prog, mach, w, dwarf, test) { }
+	
+	def unimplemented() {
+		mach.fail("unimplemented");
+	}
+
+	// Override Arm64Backend	
+
+    def genSigHandlerInstall(signo: int, handler: Addr) { unimplemented(); }
```
Comment (titzer) [inline aeneas/src/arm64/Arm64Linux.v3:34]: Formatting is off a little here.

**Review by titzer — APPROVED** [review]
lgtm other nit. Can you please run `aeneas bump` to update the version number?

**Comment by YonahGoldberg** [comment]
add00 and add01 are working:

Screenshot 2024-05-17 at 5 14 36 PM

**Review by titzer — APPROVED** [review]
I think this is good enough to land for now.

**Pull request merged**